lace/emulator/gp_emulator.py

```diff
@@ -95,15 +130,10 @@ def _training_points_k_bin(self,archive):
         ''' Method to get the Y training points in the form of the P1D
         at different k values '''
 
-        P1D_k=np.empty([len(self.archive.data),self.k_bin-1])
-        for aa in range(len(self.archive.data)):
-            P1D_k[aa]=self.archive.data[aa]['p1d_Mpc'][1:self.k_bin]
-            if self.reduce_var_k:
-                P1D_k[aa]*=(1+self.training_k_bins)
-            if self.reduce_var_z:
-                P1D_k[aa]*=1./((1+self.archive.data[aa]["z"])**3.8)
-            if self.reduce_var_mf:
-                P1D_k[aa]*=((self.archive.data[aa]["mF"])**2)
+        P1D_k=np.empty([len(getattr(self.archive, self.key_data)),self.k_bin-1])
+        for aa in range(len(getattr(self.archive, self.key_data))):
+            P1D_k[aa]=getattr(self.archive, self.key_data)[aa]['p1d_Mpc'][1:self.k_bin]
```

Comment:
I'm still not convinced we need this `key_data` option, but if we do need it, we should probably write  a self.get_data() function that hides all this getattr calls that are now everywhere in the code.